mountpoint-s3-client/src/object_client.rs

```diff
@@ -647,7 +682,7 @@ impl fmt::Display for ObjectAttribute {
 ///
 /// See [Checksum](https://docs.aws.amazon.com/AmazonS3/latest/API/API_Checksum.html) in the *Amazon
 /// S3 API Reference* for more details.
-#[derive(Debug)]
+#[derive(Clone, Debug)]
```

Comment:
Thoughts on having the checksums typed once #1082 goes in?

Reply:
I don't have a strong opinion. I'd be inclined to stick close to the REST API but I see we already diverged a bit by creating a `Checksum` struct in the first place.

Reply:
Thinking about this a bit more:

- Probably HeadObject shouldn't be returning this struct, but instead each checksum on the `HeadObjectResult`. (This struct is mirroring how GetObjectAttributes returns the values.)
- We could have `Option<Crc32>`, etc. here. That sounds fine to me.

Reply:
If we are talking about whether to wrap the `Option`al checksums in a `Checksum` struct, my slight preference is in favour - I don't think its really diverging from the API in meaningful ways. But I don't think it's really too interesting either way.

I care a bit more about the "typed" checksums. As I sees it we should either:
1. keep everything as `String`, neutral and close to the API. We'll offer separate facilities to parse checksums (and encode as base64).
2. consistently use our checksum types.

My preference is for (2), which seems much nicer for users.

Reply:
None of this is blocking for this PR, just bringing it up because we may want to go through all related API changes in one release cycle.